network/network-tutorial-overlay.md

```diff
@@ -328,66 +326,66 @@ The example uses Linux hosts, but the same commands work on Windows.
         ```
 
-        Otherwise, the `docker swarm join` command will time out. In this case,
-        run `docker swarm leave --force` on `node2`, verify your network and
-        firewall settings, and try again.
+    Otherwise, the `docker swarm join` command times out. If so, run `docker
+    swarm leave --force` on `host2`, verify your network and firewall
+    settings, and try again.
 
-2.  Create an attachable overlay network called `test-net` on `host1`.
+2.  On `host1`, create an attachable overlay network called `test-net`:
 
     ```bash
-    $ docker network create --driver=overlay --attachable test-net
+    (host1) $ docker network create --driver=overlay --attachable test-net
     ```
 
-    You don't need to manually create the overlay on `host2` because it will
-    be created when a container or service tries to connect to it from `host2`.
+    > On `host2`, the `test-net` overlay network is created when you run `alpine2` on it.
 
-3.  On `host1`, start a container that connects to `test-net`:
+3.  On `host1`, start an interactive container (`alpine1`), with a terminal, that connects to `test-net`:
 
     ```bash
-    (host1) $ docker run -dit \
-              --name alpine1 \
-              --network test-net \
-              alpine
+    (host1) $ docker run -it --name alpine1 --network test-net alpine
+    / #
     ```
 
-4.  On `host2`, start a container that connects to `test-net`:
+4.  On `host2`, list the available networks -- notice that `test-net` does not yet exist:
 
     ```bash
-    (host2) $ docker run -dit \
-              --name alpine2 \
-              --network test-net \
-              alpine
+    (host2) $ docker network ls
     ```
-    The `-dit` flags mean to start the container detached
-    (in the background), interactive (with the ability to type into it), and
-    with a TTY (so you can see the input and output).
-    
-    > **Note**: There is nothing to prevent you from using the same container
-    > name on multiple hosts, but automatic service discovery will not work if
-    > you do, and you will need to refer to the containers by IP address.
 
-    Verify that `test-net` was created on `host2`:
+5.  On `host2`, start a detached and interactive container (`alpine2`), without a teriminal, that
+    connects to `test-net`:
+
+    ```bash
+    (host2) $ docker run -di --name alpine2 --network test-net alpine
+    b30567dd179f819d11feb516a278b0c560399e5f53db8214ca23432431d5b6c0
+    ```
+
+    The `-d`, `-i`, `-t`
+    [flags](https://docs.docker.com/engine/reference/commandline/exec/#options)
+    start a container that is **detached** (runs in the background),
+    **interactive** (with `stdin` open), and with a **tty** (terminal). We run
+    `alpine1` interactively, and with a terminal, so that we can work within its
+    shell and ping `alpine2`. We run `alpine2` interactively, so that `alpine1`
+    can access its `stdin`/`stdout`, but we don't need a shell so we can make it
```

Comment:
> We run `alpine2` interactively, so that `alpine1` can access its `stdin`/`stdout`. This is not correct since alpine1 can't access stdin/out of alpine2.

Almost, but not quite.
You deploy alpine2 in interactive so the `/bin/sh` running on alpine2 can have its own stdin/stdout and continue running on the background. Without it, the alpine2 container would stop immediately.

I would prefer not to talk too much about these details since they distract from the main point: creating your network with `--attachable`.
All of the rest are just artifacts of the particular example we're using to illustrate `ping` working across nodes.

Reply:
I removed most of it.  Simplified.